### PR TITLE
Allows players to heatproof reinforced floors, and upgrades certain floors to be heatproof

### PR DESCRIPTION
**Pull request by Citinited**
Ever wondered why your fire was cooling itself seemingly for no reason? Well now you can stop that!

Use a sheet of plasteel on a reinforced floor and it will fully heatproof that floor. I've also updated floors in the turbine, toxins burn chamber, and the incinerator to be heatproof.

:cl:
rscadd: You can now fully insulate reinforced floors using a sheet of plasteel.
tweak: Certain floors in the turbine, incinerator, and toxins are now thermally insulated.
/:cl:

EDIT: this buffs the TEG and turbines a little bit; this might affect your opinion on this change.

### CODE REVIEW

File: code/game/turfs/simulated/floor/plating.dm [hunk 1]
```diff
@@ -106,6 +107,16 @@
 			new /obj/item/stack/rods(src, 2)
 			ChangeTurf(/turf/simulated/floor/plating)
 			return
+	if(istype(C, /obj/item/stack/sheet/plasteel) && insulated == 0) //Insulating the floor
```
Comment (tigercat2000) [inline code/game/turfs/simulated/floor/plating.dm]: Don't use `insulated == 0`, use `!insulated`

File: code/game/turfs/simulated/floor/plating.dm [hunk 2]
```diff
@@ -106,6 +107,16 @@
...
+		to_chat(user, "<span class='notice'>You begin insulating [src]...</span>")
+		if(do_after(user, 40, target = src) && insulated == 0) //You finish insulating the insulated insulated insulated insulated insulated insulated insulated insulated vacuum floor
```
Comment (tigercat2000) [inline code/game/turfs/simulated/floor/plating.dm]: Same here.

File: code/game/turfs/simulated/floor/plating.dm [hunk 3]
```diff
@@ -106,6 +107,16 @@
...
+			to_chat(user, "<span class='notice'>You finish insulating [src].</span>")
+			var/obj/item/stack/sheet/plasteel/W = C
+			W.use(1)
+			thermal_conductivity = 0.0
```
Comment (tigercat2000) [inline code/game/turfs/simulated/floor/plating.dm]: Just change this to 0, BYOND has no difference between ints and floats, they're all floats underneath

File: code/game/turfs/simulated/floor/plating.dm [hunk 4]
```diff
@@ -161,7 +172,17 @@
 	nitrogen = 0
 	temperature = TCMB
 
+/turf/simulated/floor/engine/insulated
+	name = "insulated reinforced floor"
+	icon_state = "engine"
+	insulated = 1
+	thermal_conductivity = 0.0
```
Comment (tigercat2000) [inline code/game/turfs/simulated/floor/plating.dm]: Same thing here, just a 0

Comment (Citinited) [inline code/game/turfs/simulated/floor/plating.dm]: Updated all, thanks for the review!

**Comment by Anticept** [comment]
I'm not sure if I can support this. On one hand, infinite power TEGs and Turbines. On the other... infinite power tegs and turbines.

But yeah, to be serious, something just feels wrong about making things completely heatproof. In addition, it's not terribly difficult for science to heatproof the floor in toxins, which will then make it MUCH EASIER to mass produce maxcap bombs.

**Comment by lolexploit** [comment]
@Anticept
You can make 6 max cap toxin bombs in about 15 minutes since roundstart without even using a burnchamber, just by utilizing heater and freezer machines. Even ignoring this fact, there's no physical relation between extra heatproofing any part of toxins and speed of bomb making.

**Comment by Citinited** [comment]
@Anticept Walls are already completely heatproof, so this change brings floors in line with that.

As for maxcaps, a burn chamber is completely unnecessary to make them.

**Comment by Citinited** [comment]
I'm okay with buffing the TEG / turbine, to be honest. In any case, if you're using those engines then you're already dumping a lot of heat into space (and adding cold fuel to the turbine), which means that these engines *shouldn't* go infinite.

It is something to bear in mind, though. I shall make a note in my OP about it.

**Comment by Anticept** [comment]
Actually, you don't need to dump heat for the turbine. You can literally just turn on the input and let it flow. There's no pressure limit that I'm aware of.

**Comment by Citinited** [comment]
I mean the turbine takes heat in and exhausts it out the back.

**Comment by Anticept** [comment]
After a discussion on github, I realized I failed at thermodynamics and forgot the incoming air also needs to be heated up. I blame it on the fact I've been up for 28 hours as of now.

Got my thumbs up here.

**Comment by AndriiYukhymchak** [comment]
If TEG/Turbine gets too out of control, I will balance it then with magic of math (by lowering effectiveness on high settings). This will at least require multiple tegs/turbines for same result

**Comment by SamHPurp** [comment]
I fully support this.

**Comment by DesolateG** [comment]
Don't touch my damn TEG.

**Comment by Citinited** [comment]
Oh yeah, I should probably add metastation to this PR

Or they can do it themselves with their 120+ population

**Comment by Citinited** [comment]
Metastation flooring upgraded, please update labels @Crazylemon64

**Comment by Fox-McCloud** [comment]
This is what engine floors already exist for, which are build by using rods on regular plating.

This is is going to cause all sorts of silly behavior; something with a thermal conductivity of 0 is going to gain heat infinitely, as no temperature will ever be radiated from it.

@tigercat2000 you're more familiar with the inner mathematical workings of LINDA, but, if I'm reading this correctly, `thermal_conductivity` being 0 is going to lead to all sorts of zaney temperature behaviors.

**Comment by Citinited** [comment]
@Fox-McCloud Regular engine floors leak heat, which is annoying if you're trying to do anything which isn't just "pump as much fuel into the chamber as possible". For example, if you're trying to make a fire with a very low fuel input, then your fire isn't going to heat up very much as most of that heat will be conducted through the floors.

The temperature will not go infinite because: a) the fire needs to heat up the incoming fuel, which reduces the temperature of the fire and means your fire will eventually reach an equilibrium; and b) you are limited by how much fuel you can inject into the fire (which prevents you getting truly insane fires.

Finally, those zaney temperatures shouldn't happen because only this specific type of floor is thermally insulated -- if your fire spreads out onto normal tiles, it will begin to cool normally.

**Comment by tigercat2000** [comment]
@Fox-McCloud Non-true values of thermal_conductivity are treated specially in LINDA, they basically can't leak off temperature in any conditions. New plasma introduced to the turf really won't do fuck all to the temperature unless it's a single-tile chamber.

**Pull request merged**